File: clients/php/examples/quickstart.php
<?php

/**
 * Priostack ACN — PHP quickstart.
 *
 * register -> connect -> createSpace -> store -> fetch, printing results.
 *
 * Run it:
 *     composer install                       # generates vendor/autoload.php (no 3rd-party deps)
 *     php examples/quickstart.php "my-agent"
 *
 * The endpoint can be overridden with the PRIOSTACK_ENDPOINT environment
 * variable; the display name is the first CLI argument (default "php-sdk-smoke").
 *
 * All network calls live inside main(), so requiring this file (or the library)
 * never fires a request on its own.
 */

declare(strict_types=1);

namespace Priostack\Acn\Examples;

use Priostack\Acn\Client;
use Priostack\Acn\Exception\AcnError;

// Prefer the Composer autoloader; fall back to a minimal PSR-4 loader so the
// example also runs from a bare clone (the library has no third-party deps).
$autoload = __DIR__ . '/../vendor/autoload.php';
if (is_file($autoload)) {
    require $autoload;
} else {
    spl_autoload_register(static function (string $class): void {
        $prefix = 'Priostack\\Acn\\';
        if (!str_starts_with($class, $prefix)) {
            return;
        }
        $relative = str_replace('\\', '/', substr($class, strlen($prefix)));
        $file = __DIR__ . '/../src/' . $relative . '.php';
        if (is_file($file)) {
            require $file;
        }
    });
}

/**
 * @param list<string> $argv
 */
function main(array $argv): int
{
    $displayName = $argv[1] ?? 'php-sdk-smoke';
    $endpoint = getenv('PRIOSTACK_ENDPOINT') ?: Client::DEFAULT_ENDPOINT;

    $acn = new Client($endpoint);

    try {
        $reg = $acn->register($displayName);
        printf("registered   agentId=%s  tier=%s\n", $reg->agentId, $reg->tier);

        $conn = $acn->connect();
        printf("connected    session=%s  maxTokens=%d\n", $conn->sessionId, $conn->resolvedMaxTokens);

        $space = $acn->createSpace('php-quickstart-memory');
        printf("created      space=%s\n", $space->spaceId);

        $stored = $acn->store($space->spaceId, [
            ['content' => 'Refunds over $500 need manager approval.', 'type' => 'declaration'],
            ['content' => 'Checkout p99 latency was 812ms on 2026-09-09.', 'type' => 'measurement'],
        ]);
        printf("stored       %d object(s)\n", $stored->count());

        $hits = $acn->fetch($space->spaceId, 'refund');
        printf("fetched      matched=%d of total=%d  (returnedTokens=%d)\n", $hits->matched, $hits->total, $hits->returnedTokens);
        foreach ($hits->objects as $obj) {
            printf("  - [%s] %s\n", (string) ($obj['kind'] ?? '?'), (string) ($obj['content'] ?? ''));
        }

        return 0;
    } catch (AcnError $e) {
        fwrite(STDERR, sprintf("ACN error (%s): %s\n", $e->method ?? '?', $e->getMessage()));

        return 1;
    } finally {
        $acn->close();
    }
}

exit(main($argv));

File: clients/rust/examples/quickstart.rs
//! Quickstart: register -> connect -> create_space -> store -> fetch.
//!
//! All live network calls run inside `main`, so building the library does not
//! fire any requests. Run it against the live endpoint with:
//!
//! ```text
//! cargo run --example quickstart
//! ```
//!
//! Optional environment overrides:
//! * `PRIOSTACK_ENDPOINT`      — RPC endpoint (default https://priostack.com/mcp)
//! * `PRIOSTACK_DISPLAY_NAME`  — agent display name (default rust-sdk-quickstart)

use std::error::Error;

use priostack_acn::{Client, Object, DEFAULT_ENDPOINT};

fn main() -> Result<(), Box<dyn Error>> {
    let endpoint = std::env::var("PRIOSTACK_ENDPOINT").unwrap_or_else(|_| DEFAULT_ENDPOINT.to_string());
    let display_name =
        std::env::var("PRIOSTACK_DISPLAY_NAME").unwrap_or_else(|_| "rust-sdk-quickstart".to_string());

    let acn = Client::builder().endpoint(endpoint).build()?;

    let reg = acn.register(&display_name)?;
    println!(
        "registered agent {} (account {}, tier {})",
        reg.agent_id, reg.account_id, reg.tier
    );

    let conn = acn.connect()?;
    println!(
        "connected session {} (account {})",
        conn.session_id, conn.resolved_account
    );

    let space = acn.create_space("prod-memory", &[])?;
    println!("created space {}", space.space_id);

    let stored = acn.store(
        &space.space_id,
        &[
            Object::declaration("Refunds over $500 need manager approval."),
            Object::observation("Customer 4417 asked about EU shipping twice this week."),
        ],
    )?;
    println!("stored {} objects", stored.count());

    let hits = acn.fetch(&space.space_id, "refund", 0)?;
    println!(
        "fetched {} of {} objects (query = \"refund\"):",
        hits.objects.len(),
        hits.total
    );
    for content in hits.contents() {
        println!("  - {content}");
    }

    acn.disconnect()?;
    println!("disconnected");

    Ok(())
}
